libr/bin/format/mach0/mach0.c

```diff
@@ -3263,10 +3266,45 @@ static int parse_import_ptr(struct MACH0_(obj_t) *bin, struct reloc_t *reloc, in
 	return false;
 }
 
-struct import_t *MACH0_(get_imports)(struct MACH0_(obj_t) *bin) {
+static RBinImport *import_from_name(RBin *rbin, const char *orig_name) {
+	RBinImport *ptr = R_NEW0 (RBinImport);
+	if (!ptr) {
+		return NULL;
+	}
+
+	char *name = (char*) orig_name;
+	const char *_objc_class = "_OBJC_CLASS_$";
+	const char *_objc_metaclass = "_OBJC_METACLASS_$";
+	const char * type = "FUNC";
+
+	if (r_str_startswith (name, _objc_class)) {
+		name += strlen (_objc_class);
+		type = "OBJC_CLASS";
+	} else if (r_str_startswith (name, _objc_metaclass)) {
+		name += strlen (_objc_metaclass);
+		type = "OBJC_METACLASS";
+	}
+
+	// Remove the extra underscore that every import seems to have in Mach-O.
+	if (*name == '_') {
+		name++;
+	}
+	ptr->name = r_str_ndup (name, R_BIN_MACH0_STRING_LENGTH - 1);
+	ptr->bind = "NONE";
+	ptr->type = r_str_constpool_get (&rbin->constpool, type);
+	return ptr;
+}
+
+const RPVector *MACH0_(load_imports)(RBinFile *bf, struct MACH0_(obj_t) *bin) {
 	r_return_val_if_fail (bin, NULL);
+	if (bin->imports_loaded) {
+		return &bin->imports_cache;
+	}
+
+	bin->imports_loaded = true;
+	r_pvector_init (&bin->imports_cache, (RPVectorFree) r_bin_import_free);
+	r_pvector_reserve (&bin->imports_cache, bin->dysymtab.nundefsym);
```

Comment:
use a local variable like `int amount = bin->dysymtab.nundefsym` . so you can check if the value is invalid (0 will cause invalid allocations and large value can be causing negative operations or invalid allocation because not enough memory if its too large).

having the var in local instead of reusing bin->dysymtab.nundefsym:
* allow to dim/trim to a decent value
* detach from the bin header information
* faster access because this is used in the for loop and you dont need to dereference pointers to know the value.